docs/quickstarts/python-quickstart.md

```diff
@@ -0,0 +1,212 @@
+quickstart example, we've already generated the server and client
+stubs from [helloworld.proto](https://github.com/grpc/grpc/tree/{{site.data.config.grpc_release_branch}}/examples/protos/helloworld.proto),
+but you'll need the tools for the rest of our quickstart, as well as later
+tutorials and your own projects.
+
+To install gRPC tools, run:
+
+```sh
+pip install grpcio-tools
+```
+
+## Download the example
+
+You'll need a local copy of the example code to work through this quickstart. Download the example code from our Github repository:
+
+```sh
+  $ # Clone the repository to get the example code:
+  $ git clone https://github.com/grpc/grpc
+  $ # Navigate to the "hello, world" Python example:
+  $ cd grpc/examples/python/helloworld
+```
+
+## Run a gRPC application
+
+From the `examples/python/helloworld` directory:
+
+1. Run the server
+
+   ```sh
+   $ python2.7 greeter_server.py
+   ```
+
+2. In another terminal, run the client
+
+   ```sh
+   $ python2.7 greeter_client.py
+   ```
+
+Congratulations! You've just run a client-server application with gRPC.
+
+## Update a gRPC service
+
+Now let's look at how to update the application with an extra method on the
+server for the client to call. Our gRPC service is defined using protocol
+buffers; you can find out lots more about how to define a service in a `.proto`
+file in [What is gRPC?]() and [gRPC Basics: Python][]. For now all you need
+to know is that both the server and the client "stub" have a `SayHello` RPC
+method that takes a `HelloRequest` parameter from the client and returns a
+`HelloResponse` from the server, and that this method is defined like this:
+
+
+```
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+Let's update this so that the `Greeter` service has two methods. Edit `examples/proto/helloworld.proto` and update it with a new `SayHelloAgain` method, with the same request and response types:
+
+```
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+  // Sends another greeting
+  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+(Don't forget to save the file!)
+
+## Generate gRPC code
+
+Next we need to update the gRPC code used by our application to use the new service definition. From the `examples/python/helloworld` directory:
+
+```
+$ python2.7 run_codegen.py
+```
+
+This script invokes the protocol buffer compiler as:
```

Comment:
These words made more sense when `run_codegen.sh`'s contents were exactly the same as the following line. Now that we're providing `run_codegen.py`, the contents of which call a Python function rather than invoke the Python interpreter, I think this line of text should be changed to something like "This script has the same effect as:".